### PR TITLE
Feature/test region allocation

### PR DESCRIPTION
**Pull request by jeremy-large**
SImple, and somewhat fragile, tests, around regional analysis code

### CODE REVIEW

**Review by blackbox-tech — COMMENTED** [review]
General comment (beyond the scope of this PR):
```test=False``` is bad.
The function should take a config that points it to a test or production data file.  The production data is likely to be update and this should not break the unit-tests.

File: lib/codit/population/networks/home_locations.py [hunk 1]
```diff
@@ -210,7 +213,8 @@ def build_households_home_list(district_type=DEFAULT_DISTRICT_TYPE):
         list_households_info += allocate_homes_to_district(pop_district[2] / AVERAGE_HOUSEHOLD_SIZE, tmp_coords_district)
 
     df_home_list = pd.DataFrame(list_households_info, columns=df_coordinates.columns)
-    df_home_list.to_csv(FULL_HOME_LIST_CSV, index=False)
```
Comment (blackbox-tech) [inline lib/codit/population/networks/home_locations.py:213]: General comment (beyond the scope of this PR):
You should not have hardcoded paths in functions, this file path should be an argument or config.
I'll can add a PR to fix this in the future.

File: lib/codit/population/networks/regions.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ class LSOA(Place):
     """
     with smart_open.open(POPULATION_LSOA_CSV) as fh:
         LSOAs = pd.read_csv(fh)
```
Comment (blackbox-tech) [inline lib/codit/population/networks/regions.py:58]: General comment, this is a bit confusing as LSOA is a variable. PEP states:
* variables should be lowercase (snake_case)
* class types should be start with an upper case letter (CamelCase)
* constants should be UPPERCASE.

**Review by blackbox-tech — APPROVED** [review]
Generally I think this is fine.  I can do another PR which you can review which will:
* Remove hardcoded paths and introduce extensible config
* Use pathlib instead of os.path
* Clean up the file structure and compress data (eg have a leeds folder for Leeds specific data)
* Have test data specifically for unit-tests (not in the share directory so it does not get packaged).

File: lib/codit/population/networks/home_locations.py (hunk 1 above)
Comment (jeremy-large) [reply lib/codit/population/networks/home_locations.py:213]: Noted, thank you

**Pull request merged**